proto/trip.proto

```diff
@@ -245,15 +255,15 @@ message TripLeg {
 
     optional Edge edge = 1;
     repeated IntersectingEdge intersecting_edge = 2;
-    optional double elapsed_time = 3;        // seconds from start of leg
-    optional uint32 admin_index = 4;         // index into the admin list
-    optional Type type = 5;                  // The type of node
-    optional bool fork = 6;                  // Fork
-    optional TransitPlatformInfo transit_platform_info = 7;
-    optional TransitStationInfo transit_station_info = 8;
-    optional TransitEgressInfo transit_egress_info = 9;
-    optional string time_zone = 10;
-    optional double transition_time = 11;    // seconds spent transitioning through this node
+    optional uint32 admin_index = 3;         // index into the admin list
+    optional Type type = 4;                  // The type of node
+    optional bool fork = 5;                  // Fork
+    optional TransitPlatformInfo transit_platform_info = 6;
+    optional TransitStationInfo transit_station_info = 7;
+    optional TransitEgressInfo transit_egress_info = 10;
+    optional string time_zone = 11;
+    optional PathCost cost = 12;             // how much cost did it take at this node in the path
+    repeated PathCost recosts = 13;          // how much cost did it take at this node in the path for optional recostings
```

Comment:
i could have left the numbers unchanged here but i really wanted to get rid of transition_time and elapsed_time as they are concepts taht no longer make sense with the full cost object being stored. we wouldnt really have been able to avoid the pinpoint tests failing because they rely on the elapsed_time at least to some degree.